scripts/tasks.py

```diff
@@ -44,3 +45,20 @@ def opt_distance(config):
                                  plot=os.path.join(setup.root_dir, task.plot))
     logger.info(f'Detector distance inferred from powder rings: {dist} mm')
     logger.debug('Done!')
+
+def find_peaks(config):
+    setup = config.setup
+    task = config.find_peaks
+    """ Perform adaptive peak finding on run. """
+    taskdir = os.path.join(setup.root_dir, 'index')
+    os.makedirs(taskdir, exist_ok=True)
+    pf = PeakFinder(exp=setup.exp, run=setup.run, det_type=setup.det_type, outdir=os.path.join(taskdir ,f"r{setup.run:04}"), 
+                    tag=task.tag, mask=task.mask, psana_mask=task.psana_mask, min_peaks=task.min_peaks, max_peaks=task.max_peaks,
+                    npix_min=task.npix_min, npix_max=task.npix_max, amax_thr=task.amax_thr, atot_thr=task.atot_thr, 
+                    son_min=task.son_min, peak_rank=task.peak_rank, r0=task.r0, dr=task.dr, nsigm=task.nsigm)
+    logger.debug(f'Performing peak finding for run {setup.run} of {setup.exp}...')
+    pf.find_peaks()
+    pf.curate_cxi()
+    pf.summarize() 
+    logger.info(f'Saving CXI files and summary to {taskdir}/r{setup.run:04}')
+    logger.debug('Done!')
```

Comment:
We would want to report some statistics to the eLog Report column here. 

It could look like this:
```python
requests.post(update_url, json=[ { "key": "Number of events processed", "value": f"{n_events_per_rank[-1]}" },
                                                         { "key": "Number of hits found", "value": f"{self.n_hits_total}"}, ... ])
```

Reply:
Pass `update_url` to a PeakFinder's `self.report` function

Reply:
Function added, still needs to be tested via the elog.